data/src/main/java/org/apache/iceberg/data/BaseDeleteLoader.java

```diff
@@ -146,6 +151,26 @@ private <T> Iterable<T> materialize(CloseableIterable<T> iterable) {
   @Override
   public PositionDeleteIndex loadPositionDeletes(
       Iterable<DeleteFile> deleteFiles, CharSequence filePath) {
+    if (containsDVs(deleteFiles)) {
```

Comment:
The `containsDVs` check here seems more broad than the assumption that there is one and only one DV.  Should we make the contains check assert those expectations? (e.g. `containsSingleDV`?)

Reply:
Eduard had the same question when reviewing the prototype PR. Let me think how to make the code more obvious. Our `Iterables.getOnlyElement` below will fail if there are multiple files.